packages/utils/src/dsn.ts

```diff
@@ -1,127 +1,114 @@
-import { DsnComponents, DsnLike, DsnProtocol } from '@sentry/types';
+import { Dsn, DsnComponents, DsnLike, DsnProtocol } from '@sentry/types';
 
 import { isDebugBuild } from './env';
 import { SentryError } from './error';
 
 /** Regular expression used to parse a Dsn. */
 const DSN_REGEX = /^(?:(\w+):)\/\/(?:(\w+)(?::(\w+))?@)([\w.-]+)(?::(\d+))?\/(.+)/;
 
-/** Error message */
-const ERROR_MESSAGE = 'Invalid Dsn';
+function isValidProtocol(protocol?: string): protocol is DsnProtocol {
+  return protocol === 'http' || protocol === 'https';
+}
 
-/** The Sentry Dsn, identifying a Sentry instance and project. */
-export class Dsn implements DsnComponents {
```

Comment:
Mind writing a quick summary on how to convert from using the `Dsn` class to the functions? We’ll need to add it to the changelog for react native and electron.